administrator/components/com_associations/associations.php

```diff
@@ -15,6 +15,17 @@
 	return JError::raiseWarning(404, JText::_('JERROR_ALERTNOAUTHOR'));
 }
 
+$input     = JFactory::getApplication()->input;
+$component = $input->get('component', '', 'string');
+$splitcpnt = explode('.', $component);
+$component = $splitcpnt[0];
+
+// Check if user has permission to access the component
+if ($component != '' && !JFactory::getUser()->authorise('core.manage', $component))
+{
+    throw new Exception(JText::_('JERROR_ALERTNOAUTHOR'), 403);
+}
+
 JLoader::register('AssociationsHelper', __DIR__ . '/helpers/associations.php');
```

Comment:
@jreys  this will not work in all scenarios

something like this is more expected to work. but needs tests:
```php
JLoader::register('AssociationsHelper', __DIR__ . '/helpers/associations.php');

$component = JFactory::getApplication()->input->get('component', '', 'string');

if ($component !== '')
{
	$cp = = AssociationsHelper::getComponentProperties($component);

	if (!$cp[$key]->associations->support)
	{
		throw new Exception(JText::sprintf('The component %s doesn't support associations.', $cp->realcomponent), 404);
	}

	if (!JFactory::getUser()->authorise('core.manage', $cp->realcomponent))
	{
		throw new Exception(JText::_('JERROR_ALERTNOAUTHOR'), 403);
	}
}
```

Also you need to refactor the side by side view URL parameters so this can work there too.